### ISSUE TITLE
Issue in the train, test masks within elliptic dataset

### ISSUE THREAD

**Issue by shravankumar147** [pyg-team/pytorch_geometric#4314]
### 🐛 Describe the bug

In the Elliptic dataset, the train_mask and test mask are not as expected. Hence I wrote a test for the elliptic dataset here.

Reference : [test_elliptic_dataset](https://github.com/pyg-team/pytorch_geometric/pull/4313)

This is failing as the mask is not meeting the requirement. 

```
=========================== short test summary info ============================
FAILED test/datasets/test_elliptic.py::test_elliptic - assert 0 == (3462,)
===== 1 failed, 2068 passed, 21 skipped, 15 warnings in 168.01s (0:02:48) ======
```

Issue could be at:
```
        train_mask = (time_step) < 35 & (y != 2)
        test_mask = (time_step) >= 35 & (y != 2)
```

Looking forward for your inputs on this issue. 

### Environment

* PyG version:  2.0.4
* PyTorch version: 1.11.0+cpu
* OS: 'Windows'
* Python version: '3.10.2'
* CUDA/cuDNN version: None
* How you installed PyTorch and PyG (`conda`, `pip`, source): pip 
* Any other relevant information (*e.g.*, version of `torch-scatter`): 2.0.9
